revolutionpi/encoder.go

```diff
@@ -83,26 +86,35 @@ func newEncoder(
 	return &revolutionPiEncoder{Named: conf.ResourceName().AsNamed(), pin: enc}, nil
 }
 
-// TODO implementing in https://viam.atlassian.net/browse/RSDK-7595
 func (enc *revolutionPiEncoder) Position(ctx context.Context, positionType encoder.PositionType,
 	extra map[string]interface{},
 ) (float64, encoder.PositionType, error) {
-	return 0, encoder.PositionTypeTicks, nil
+	pos, err := enc.pin.Value()
+	if err != nil {
+		return 0, encoder.PositionTypeTicks, err
+	}
+	// rev pi encoder values are signed int32
+	signedPos := int32(pos) - enc.zeroPos.Load()
```

Comment:
Something in here (possibly the comment?) is off: if `pos` is already a signed int32, why do we need to cast it to a (signed) int32?

I wonder if the comments on lines 96 and 99 should be rolled together ("The rev pi encoder values are int32, but the encoder API expects a float64"), and the `int32(...)` call on this line should be removed.

Reply:
ohh, maybe the comment should have said it was an _unsigned_ int32? Then the code makes sense.

Reply:
yeah sorry this is kinda confusing, `Value` is returned as an unsigned int32(since the counters only count up), but the encoder returns int32(since encoders go both directions). I'll try to tweak the comments to make this clearer